### PR TITLE
credentials: clear private key when dropped

### PR DESCRIPTION
**Pull request by BusyJay**

### CODE REVIEW

File: src/credentials.rs [hunk 1]
```diff
@@ -129,12 +144,18 @@ impl ChannelCredentialsBuilder {
...
 
-    pub fn root_cert<S: Into<Vec<u8>>>(mut self, cert: S) -> ChannelCredentialsBuilder {
+    pub fn root_cert(mut self, cert: Vec<u8>) -> ChannelCredentialsBuilder {
         self.root = Some(CString::new(cert).unwrap());
         self
     }
 
-    pub fn cert<S: Into<Vec<u8>>>(mut self, cert: S, key: S) -> ChannelCredentialsBuilder {
+    pub fn cert(mut self, cert: Vec<u8>, mut key: Vec<u8>) -> ChannelCredentialsBuilder {
+        if key.capacity() == key.len() {
```
Comment (siddontang) [inline src/credentials.rs]: can we check the last char is NULL? using capacity equals length confused me.

Comment (BusyJay) [reply src/credentials.rs]: Actually, there should be no '\0' in key. Check capacity to ensure there is at least one slot to append a '\0'.

Comment (siddontang) [reply src/credentials.rs]: if the capacity is not equal length, why not still adding \0？

Comment (BusyJay) [reply src/credentials.rs]: Then the buffer is reallocated. It can't be determined if the buffer is extended in place or changed to point to a new memory.

Comment (BusyJay) [reply src/credentials.rs]: `CString::new` will handle this automatically.

**Pull request merged**